compiler/rustc_passes/src/check_attr.rs

```diff
@@ -823,21 +842,27 @@ impl CheckAttrVisitor<'tcx> {
         attrs: &[Attribute],
     ) -> bool {
         debug!("Checking target: {:?}", target);
-        if target == Target::Fn {
-            for attr in attrs {
-                if self.tcx.sess.is_proc_macro_attr(attr) {
-                    debug!("Is proc macro attr");
-                    return true;
+        match target {
+            Target::Fn => {
+                for attr in attrs {
+                    if self.tcx.sess.is_proc_macro_attr(attr) {
+                        debug!("Is proc macro attr");
+                        return true;
+                    }
                 }
+                debug!("Is not proc macro attr");
+                false
+            }
+            Target::Field | Target::Arm | Target::MacroDef => true,
```

Comment:
Please leave a FIXME so we start warning on them in the future

Reply:
It is correct to apply `#[allow_internal_unstable]` to macros, so we won't warn about it in the future, will we?

Reply:
Right, these should just be allowed directly without any additional checks.